### PR TITLE
prov/sockets: Fix coveritylocking issues

### PR DESCRIPTION
**Pull request by zachdworkin**

### CODE REVIEW

**Comment by shefty** [comment]
It's more useful if the commit header describe the problem/fix rather than a reference to a coverity report

File: prov/sockets/src/sock_wait.c [hunk 1]
```diff
@@ -137,8 +137,11 @@ static int sock_wait_wait(struct fid_wait *wait_fid, int timeout)
 			cq = container_of(list_item->fid,
 					  struct sock_cq, cq_fid);
 			sock_cq_progress(cq);
-			if (ofi_rbused(&cq->cqerr_rb))
+			pthread_mutex_lock(&cq->lock);
+			if (ofi_rbused(&cq->cqerr_rb)) {
+				pthread_mutex_unlock(&cq->lock);
 				return 1;
+			}
```
Comment (sydidelot) [inline prov/sockets/src/sock_wait.c:144]: There is a `pthread_mutex_lock(&cq->lock);` missing here, before the `break`.

**Review by sydidelot — COMMENTED** [review]
I think the PR has a bug, please check my comment.

File: prov/sockets/src/sock_wait.c (hunk 1 above)
Comment (zachdworkin) [reply prov/sockets/src/sock_wait.c]: fixed in latest push, thanks!

**Comment by j-xiong** [comment]
bot:aws:retest

**Review by j-xiong — APPROVED** [review]

**Pull request merged**